src/shogun/machine/gp/KLApproxDiagonalInferenceMethod.cpp

```diff
@@ -74,7 +74,8 @@ void CKLApproxDiagonalInferenceMethod::init()
 CKLApproxDiagonalInferenceMethod* CKLApproxDiagonalInferenceMethod::obtain_from_generic(
 		CInferenceMethod* inference)
 {
-	REQUIRE(inference, "Inference pointer not set.\n");
+	if (inference==NULL)
```

Comment:
thanks for that.

BTW there were some plans to provide obtina_from_generic for all Shogun classes automagically. Interested in doing this at some point? :)